R/utils.R

```diff
@@ -437,18 +442,122 @@ rot_x <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
     }
     angle <- 90
     vjust <- 0.5
+  } else {
+    if(is.character(hjust)) {
+      abort("hjust must be numeric or NULL in this case.")
+    }
   }
   x <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
   theme(axis.text.x=x)
 }
 
 ##' @rdname rot_x
 ##' @export
-rot_y <- function(angle=30, hjust = 1, vjust = NULL,...) {
-  y <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
+rot_y <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
+  if(isTRUE(vertical)) {
+    if(is.character(vjust)) {
+      vjust <- match.arg(vjust, c("left", "right"))
+      if(vjust=="left") vjust <- 1
+      if(vjust=="right") vjust <- 0
+    } else {
+      if(missing(vjust)) vjust <- 1
+    }
+    angle <- 90
+    hjust <- 0.5
+  } else {
+    if(is.character(vjust)) {
+      abort("`vjust` must be numeric or NULL in this case.")
+    }
+  }
+  y <- element_text(angle = angle, hjust = hjust, vjust = vjust,, ...)
   theme(axis.text.y=y)
 }
 
+
+.rotxy <- function(x, ...) UseMethod(".rotxy")
+#' @export
+.rotxy.gg <- function(x, axis = "x", ...) {
+  if(axis=="x") {
+    x <- x + rot_x(...)
+  } else {
+    x <- x + rot_y(...)
+  }
+  x
+}
+#' @export
+.rotxy.patchwork <- function(x, axis = "x", ...) {
```

Comment:
I think `patchwork` objects will get dispatched here and not the `gg` method. And `patchwork` objects will not go any farther.  Does that seem right?

Reply:
Yes, seems correct.